ui/pages/confirmations/components/confirm/blockaid-alert/blockaid-alert.tsx

```diff
@@ -2,42 +2,31 @@ import React from 'react';
 import { useSelector } from 'react-redux';
 
 import { currentConfirmationSelector } from '../../../../../selectors';
-import {
-  SecurityAlertResponse,
-  SignatureRequestType,
-} from '../../../types/confirm';
+import useSignatureSecurityAlertResponse from '../../../hooks/useSignatureSecurityAlertResponse';
+import { SignatureRequestType } from '../../../types/confirm';
 import BlockaidBannerAlert from '../../security-provider-banner-alert/blockaid-banner-alert';
 
-type SignatureSecurityAlertResponsesState = {
-  metamask: {
-    signatureSecurityAlertResponses: SecurityAlertResponse[];
-  };
-};
-
 // todo: this component can be deleted once new alert imlementation is added
 const BlockaidAlert = ({ ...props }) => {
   const currentConfirmation = useSelector(
     currentConfirmationSelector,
   ) as SignatureRequestType;
-  const signatureSecurityAlertResponses = useSelector(
-    (state: SignatureSecurityAlertResponsesState) =>
-      state.metamask.signatureSecurityAlertResponses,
+
+  const currentSecurityAlertId =
+    currentConfirmation?.securityAlertResponse?.securityAlertId;
+  const signatureSecurityAlertResponse = useSignatureSecurityAlertResponse(
```

Comment:
I think I may be misunderstanding something here, but why are we retrieving the `signatureAlertResponse` via this hook using the ID if there is already a `securityAlertResponse` on the above `currentConfirmation`?

Reply:
I was confused about this as well and was merely carrying logic over as we can see to the left. @jpuri do you have more context to this? As inquired by @matthewwalsh0, is it safe to read `securityAlertResponse` directly from `currentConfirmation` rather than querying it from `signatureSecurityAlertResponses`?

Reply:
So the main reason for it is that `SignatureController` does not have a method to update the signature data. So while it was possible to update the transaction metadata when the ppom resolves, to solve [fixing ppom blocking a confirmation page to be rendered before it resolves](https://github.com/MetaMask/metamask-extension/pull/22778), Dan was creative and for signatures stored the ppom resolution in the `appStateController` as `signatureSecurityAlertResponses`.
The loading state with the initial `securityAlertResponseId` exists on the `unapprovedMessage` as it is added on the req at the middleware layer. On the UI, for signatures, we then need to check the security alert id from the message and then get the result for that from the `signatureSecurityAlertResponses` (from the appStateController).

I know.. it is a mess. We should add a method in the SignatureController to allow updating unapprovedMessages. Then we could get rid of all this logic. But this is not in the scope of this PR.

Reply:
I see, so the `securityAlertResponse` on the confirmation is the loading response, not the final one.

This shouldn't need a `SignatureController` refactor, as ideally we shouldn't be using the raw controller objects for the current confirmation object, plus the nature of a selector is the data should automatically be updated.

It would be easier to maintain and more decoupled if we define a consistent shape for a current confirmation, then we could empower the current confirmation selector itself to build them accordingly regardless of the source, and store any custom data in a specific property similar to the approval requests which use `requestData`.

This would mean all our `currentConfirmation` objects would have a consistent format and an always up to date `securityAlertResponse`.

But of course, a future refactor.